### ISSUE TITLE
Authentication type parameter seems to accept too many values

### ISSUE THREAD

**Issue by rzhao271** [tediousjs/tedious#1406]
I noticed the allowed values for the authentication type parameter are being filtered down during runtime, but the type itself seems to accept string values, and even `undefined`. 

I'm wondering whether the type of the `authentication.type` parameter can instead be set to something like 
```ts
type AuthenticationType = 'default' | 'ntlm' | 'azure-active-directory-password' | 'azure-active-directory-access-token' | 'azure-active-directory-msi-vm' | 'azure-active-directory-msi-app-service' | 'azure-active-directory-service-principal-secret'
// ...
type: AuthenticationType;
```

This way, users would know exactly which values are valid for a given version of tedious.

![A screenshot showing the discrepancy](https://user-images.githubusercontent.com/7199958/154327726-0f30eeb6-0930-4702-9f33-8df98b102d38.PNG)

**Comment by MichaelSun90** [comment]
Hi @rzhao271, I am wondering which version of tedious driver are you on? I just checked our latest code and seems we already handled it. You can check the latest source at line 477 code [here](https://github.com/tediousjs/tedious/blob/master/src/connection.ts#:~:text=type%3F%3A%20AuthenticationType%3B). From there you can check the definition of the auth type and corresponding type handlings.

**Comment by rzhao271** [comment]
Looks like I missed those lines. I'm using https://www.npmjs.com/package/@types/tedious because I noticed https://www.npmjs.com/package/tedious had a `DT` annotation.

**Comment by MichaelSun90** [comment]
I do not think you have to use that since we do support our own types within tedious.  You can give it a try, see if it meet your needs. Our project is [here](https://github.com/tediousjs/tedious). Also, you can try to post this issue on their [board](https://github.com/DefinitelyTyped/DefinitelyTyped/issues), see if they can help you on this. 
Hi @arthurschreiber, I am not sure why there is a DT mark on the npm page for tedious, have we used their type definition before we fully migrate to typescript?

**Comment by Fuco1** [comment]
When I install your project to a typescript project, there is no typing.  The code installed from npm is javascript only, so the type declarations are missing.

**Comment by MichaelSun90** [comment]
Hi @Fuco1, Thank you for bring this up. We are using babel to compile the typescript, and seems it does not handles the type declarations. For generate the declarations , tsc is needed. 
Hi @arthurschreiber , did we not generate it for a purpose or is just because of using babel. For type declarations, should we want to set something for tsc to generate it?
